feature-libs/quote/components/items/quote-items.component.ts

```diff
@@ -5,50 +5,84 @@
 } from '@spartacus/cart/base/root';
 import { UserIdService } from '@spartacus/core';
-import { Quote, QuoteFacade } from '@spartacus/quote/root';
+import { QuoteFacade } from '@spartacus/quote/root';
 import { ICON_TYPE } from '@spartacus/storefront';
-import { NEVER, Observable, Subscription, of, zip } from 'rxjs';
-import { switchMap, tap } from 'rxjs/operators';
+import { Observable, Subscription, combineLatest, of, zip } from 'rxjs';
+import { filter, map, switchMap, tap } from 'rxjs/operators';
 import { QuoteItemsComponentService } from './quote-items.component.service';
 
+interface QuoteItemsData {
+  entries: OrderEntry[] | undefined;
+  readOnly: boolean;
+}
+
 @Component({
   selector: 'cx-quote-items',
   templateUrl: './quote-items.component.html',
+  providers: [
+    AbstractOrderContextSource,
+    { provide: AbstractOrderContext, useExisting: AbstractOrderContextSource },
+  ],
 })
 export class QuoteItemsComponent {
   protected quoteFacade = inject(QuoteFacade);
   protected activeCartFacade = inject(ActiveCartFacade);
   protected multiCartFacade = inject(MultiCartFacade);
   protected quoteItemsComponentService = inject(QuoteItemsComponentService);
   protected userIdService = inject(UserIdService);
+  protected abstractOrderContextSource = inject(AbstractOrderContextSource);
 
-  quoteDetails$: Observable<Quote> = this.quoteFacade.getQuoteDetails();
-  quoteCartReadOnly$: Observable<Cart> = this.userIdService.takeUserId().pipe(
-    switchMap((userId) => zip(this.quoteDetails$, of(userId))),
-    tap(([quote, userId]) => {
-      if (quote.cartId !== undefined) {
-        this.multiCartFacade.loadCart({
-          userId,
-          cartId: quote.cartId,
-          extraData: { active: false },
-        });
-      }
-    }),
-    switchMap(([quote, _userId]) => {
-      return quote.cartId !== undefined
-        ? this.multiCartFacade.getCart(quote.cartId)
-        : NEVER;
-    })
-  );
+  quoteItemsData$: Observable<QuoteItemsData> = this.userIdService
+    .takeUserId()
+    .pipe(
+      switchMap((userId) =>
+        combineLatest([this.quoteFacade.getQuoteDetails(), of(userId)])
+      ),
+      tap(([quote, userId]) => {
+        if (quote.cartId && !quote.isEditable) {
+          this.multiCartFacade.loadCart({
+            userId,
+            cartId: quote.cartId,
+            extraData: { active: false },
+          });
+        }
+      }),
+      switchMap(([quote, _userId]) => {
+        if (!quote.cartId) {
+          this.abstractOrderContextSource.id$.next(quote.code);
+          this.abstractOrderContextSource.type$.next(AbstractOrderType.QUOTE);
+          return zip(of(quote), of(true));
+        } else if (!quote.isEditable) {
+          this.abstractOrderContextSource.id$.next(quote.cartId);
+          this.abstractOrderContextSource.type$.next(
+            AbstractOrderType.SAVED_CART
+          );
+          return combineLatest([
+            this.multiCartFacade.getCart(quote.cartId),
```

Comment:
question: don't you want to wait until the cart is stable (I can see you've tried (re)loading it in the code above)? I mean `this.multiCartFacade.isStable(quote.cartId)`? Just asking from top of my head, not sure if it matters in your case.

Perhaps similar question applies to line 77 with `getActive()` but not sure... from it's source I can see that any emitted value is already filtered by "isStable" internally - https://github.com/SAP/spartacus/blob/7b6dc7c3cef6dd617859ce1190fce654995620e9/feature-libs/cart/base/core/facade/active-cart.service.ts#L107

Reply:
Yes you are right, multiCartFacade.getCart(quote.cartId) just returns from the store. So it would be good to only render the page when the cart is stable.

For cartFacade.getActive(): The internal check that you mention does not block the emission of unstable carts. Still the quote component receives only one emission of the cart, probably because other components are already requesting it and the activeCartService uses shareReply. 
In any case I will add the check for stable here as well